mm2src/coins/z_coin.rs

```diff
@@ -975,8 +975,11 @@ impl MarketCoinOps for ZCoin {
         )
     }
 
-    fn tx_enum_from_bytes(&self, bytes: &[u8]) -> Result<TransactionEnum, String> {
-        ZTransaction::read(bytes).map(|tx| tx.into()).map_err(|e| e.to_string())
+    fn tx_enum_from_bytes(&self, bytes: &[u8]) -> Result<TransactionEnum, MmError<TransactionErr>> {
+        match ZTransaction::read(bytes) {
```

Comment:
I would personally prefer using
```rust
        ZTransaction::read(bytes)
            .map(TransactionEnum::from)
            .map_to_mm(|e| TransactionErr::InvalidTx(e.to_string()))
```

But it's up to you whether to accept this change :slightly_smiling_face: